File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/responses/comment/CommentListResponse.java
package com.api.nature_harvest_backend.responses.comment;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@AllArgsConstructor
@Data
@Builder
@NoArgsConstructor
public class CommentListResponse {
    private List<CommentResponse> comments;
    private int totalPages;
    private long totalReviews;
    private long fiveStarReviews;
    private long fourStarReviews;
    private long threeStarReviews;
    private long twoStarReviews;
    private long oneStarReviews;
    private long imageReviews;
}

File: nature-harvest-frontend/src/app/responses/comment/comment-list.response.ts
import { CommentResponse } from './comment.response';

export class CommentListResponse {
  comments: CommentResponse[];
  totalPages: number;
  totalReviews: number;
  fiveStarReviews: number;
  fourStarReviews: number;
  threeStarReviews: number;
  twoStarReviews: number;
  oneStarReviews: number;
  imageReviews: number;

  constructor(data: any) {
    this.comments = data.comments;
    this.totalPages = data.totalPages;
    this.totalReviews = data.totalReviews;
    this.fiveStarReviews = data.fiveStarReviews;
    this.fourStarReviews = data.fourStarReviews;
    this.threeStarReviews = data.threeStarReviews;
    this.twoStarReviews = data.twoStarReviews;
    this.oneStarReviews = data.oneStarReviews;
    this.imageReviews = data.imageReviews;
  }
}
